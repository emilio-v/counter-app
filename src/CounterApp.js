import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
    const [counter, setCounter] = useState(value);

    return (
        <div id="counter-component">
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <div>
                <button onClick={() => setCounter(counter - 1)}>-1</button>
                <button onClick={() => setCounter(value)}>reset</button>
                <button onClick={() => setCounter(counter + 1)}>+1</button>
            </div>
        </div>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number,
};

export default CounterApp;
