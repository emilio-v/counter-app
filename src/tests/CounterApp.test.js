import React from 'react';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('<CounterApp>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test('it should show CounterApp', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('it should renders a default value of 10', () => {
        expect(wrapper.find('h2').text()).toBe('10');
    });

    test('it should increase the counter', () => {
        expect(wrapper.find('h2').text()).toBe('10');
        wrapper.find('button').at(2).simulate('click');
        expect(wrapper.find('h2').text()).toBe('11');
    });

    test('it should decrease the counter', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(wrapper.find('h2').text()).toBe('9');
    });

    test('it should reset the count', () => {
        const value = 0;
        const wrapper = shallow(<CounterApp value={value} />);
        wrapper.find('button').at(2).simulate('click');
        expect(wrapper.find('h2').text()).toBe(`${value + 1}`);
        wrapper.find('button').at(1).simulate('click');
        expect(wrapper.find('h2').text()).toBe(`${value}`);
    });
});
