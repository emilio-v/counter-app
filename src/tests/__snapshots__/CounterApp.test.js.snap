// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp> it should show CounterApp 1`] = `
<div
  id="counter-component"
>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <div>
    <button
      onClick={[Function]}
    >
      -1
    </button>
    <button
      onClick={[Function]}
    >
      reset
    </button>
    <button
      onClick={[Function]}
    >
      +1
    </button>
  </div>
</div>
`;
